### PR TITLE
Invalid parameters or help display did not succeed

### PR DESCRIPTION
**Pull request by phracek**
Signed-off-by: Petr "Stone" Hracek <phracek@redhat.com>

This pull request fixes two long terms pull request https://github.com/sclorg/scl-utils/pull/10 and https://github.com/sclorg/scl-utils/pull/21

### CODE REVIEW

File: shell/scl_source [hunk 1]
```diff
@@ -7,9 +7,9 @@ Don't use this script outside of SCL scriptlets!
 Options:
     -h, --help    display this help and exit"
 
-if [ $# -eq 0 -o $1 = "-h" -o $1 = "--help" ]; then
```
Comment (praiskup) [inline shell/scl_source]: The `||` is probably useless change.  You only need the double quotes.

File: shell/scl_source [hunk 2]
```diff
@@ -7,9 +7,9 @@ Don't use this script outside of SCL scriptlets!
...
+if [ $# -eq 0 ] || [ -o "$1" = "-h" ] || [ -o "$1" = "--help" ]; then
     echo "$_scl_source_help"
-    return 0
+    return 1
```
Comment (praiskup) [inline shell/scl_source]: This is actually API change.  The `scl_source` command can fail `set -e` scripts now if someone forgot to add arguments, e.g. by `scl_source "${@}"` (if there's no argument) => before it was just no-op.

File: shell/scl_source [hunk 3]
```diff
@@ -7,9 +7,9 @@ Don't use this script outside of SCL scriptlets!
 Options:
     -h, --help    display this help and exit"
 
-if [ $# -eq 0 -o $1 = "-h" -o $1 = "--help" ]; then
+if [ $# -eq 0 ] || [ -o "$1" = "-h" ] || [ -o "$1" = "--help" ]; then
```
Comment (phracek) [inline shell/scl_source]: There is a typo `-o` should not be there;)

File: shell/scl_source (hunk 1 above)
Comment (phracek) [reply shell/scl_source]: Brackets are to read, but this is a cosmetical issue. I will change it back to `-o` with double-quotes.

File: shell/scl_source (hunk 2 above)
Comment (phracek) [reply shell/scl_source]: Good point. Thanks

**Comment by phracek** [comment]
@praiskup Now the change is really small ;)

File: shell/scl_source (hunk 2 above)
Comment (phracek) [reply shell/scl_source]: This was changed based on this comment https://github.com/sclorg/scl-utils/pull/10#discussion_r101695544

Comment (praiskup) [reply shell/scl_source]: Yes, I was wrong before, I think.

**Pull request merged**